docs/src/main/asciidoc/security-keycloak-admin-client.adoc

```diff
@@ -85,16 +87,72 @@ public class RolesResource {
 }
 ----
 
-Now, you can produce the `Keycloak` bean yourself, or you can instruct Quarkus to produce `Keycloak` bean.
+<1> Create  a default Keycloak Admin Client which can perform Keycloak `master` realm administration tasks as an `admin-cli` client, such as adding new realms, clients and users.
+
+The only configuration which is required to create this Keycloak Admin Client is a Keycloak server URL, for example:
+
+[source,properties]
+----
+quarkus.keycloak.admin-client.server-url=http://localhost:8081
+----
+
+[NOTE]
+====
+It is important that `quarkus.keycloak.admin-client.server-url` is configured if you would like to have `Keycloak` injected. The injection will fail if you will attempt to inject `Keycloak` without configuring this property.
+====
+
+Injecting Keycloak Admin Client instead of creating it directly in the application code is a simpler and more flexible option but you can create the same admin client manually if necessary:
+
+[source,java]
+----
+package org.acme.keycloak.admin.client;
+
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.representations.idm.RoleRepresentation;
 
-NOTE: If you produce the bean yourself, please remember that `org.keycloak.admin.client.Keycloak` implements `AutoCloseable` and it's important to avoid resource exhaustion and related exceptions. If you configure your beans as explained below, Quarkus will do it for you.
+import javax.annotations.PostConstruct;
+import javax.annotations.PreConstruct;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.List;
+
+@Path("/api/admin")
+public class RolesResource {
+
+        Keycloak keycloak;
+
+        @PostConstruct
+        public void initKeycloak() {
+            keycloak = KeycloakBuilder.builder()
+                .serverUrl("http://localhost:8081")
+                .realm("master")
+                .clientId("admin-cli")
+                .username("admin")
+                .password("admin")
+                .build();
+        }
+
+        @PreDestroy
+        public void closeKeycloak() {
+            keycloak.close();
+        }
+
+        @GET
+        @Path("/roles")
+        public List<RoleRepresentation> getRoles() {
+            return keycloak.realm("quarkus").roles().list();
+        }
+
+}
+----
+
+You can configure Keycloak admin client to administer other realms and clients. It can use either a `password` or `client_credentials` grant to acquire an access token to call Admin REST API which requires authorization.
```

Comment:
```suggestion
You can configure Keycloak Admin Client to administer other realms and clients. It can use either a `password` or `client_credentials` grant to acquire an access token to call the Admin REST API which requires authorization.
```